daemon/monitor.go

```diff
@@ -54,6 +54,9 @@ func (daemon *Daemon) StateChanged(id string, e libcontainerd.StateInfo) error {
 			defer autoRemove()
 		}
 
+		stateCtr.dec("running")
```

Comment:
Technically yes, but at the same time I'm not too worried about it.
It's already much more racey than this.